### PR TITLE
public cli and docs for intial advanced restore

### PR DESCRIPTION
**Pull request by ryanfkeepers**
creates public docs and un-hides the advanced restore option cli flags.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3562

### CODE REVIEW

File: src/cli/restore/restore.go
Comment (gmatev) [inline src/cli/restore/restore.go:41]: We should have the examples be more realistic. I don't think restore of the full service to the original location is such an example. 

Would recommend restore of a specific item or folder instead.

File: src/cli/restore/restore.go
Comment (gmatev) [inline src/cli/restore/restore.go:47]: Are we allowing non-root restores? Should this be `/my_latest_restore` ?

File: src/cli/restore/restore.go
Comment (gmatev) [inline src/cli/restore/restore.go:43]:
```suggestion
# Restore all one drive folders OneDrive to a root folder named "my_latest_restore"
```

File: src/cli/restore/restore.go
Comment (gmatev) [inline src/cli/restore/restore.go:52]: Can we make this be a specific calendar example?

**Comment by gmatev** [comment]
I would also recommend adding a paragraph to the quick start flow that can cover the in-place restore after the existing non-destructive restore option. Would make things a bit more discoverable.

**Comment by gmatev** [comment]
Another thought is also adding a topic under `Usage` that is focused on `Restore Options`. This can cover the topic in a bit more detail which I think is needed.

File: src/cli/restore/restore.go
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:41]: done, added `--email-folder '/inbox'` as the selector.

File: src/cli/restore/restore.go
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:47]: destination always stores to the top level.  `my_latest` and `/my_latest` are treated as equivalent values.

File: src/cli/restore/restore.go
Comment (ryanfkeepers) [reply src/cli/restore/restore.go:52]: Certainly

File: src/cli/restore/restore.go
Comment (gmatev) [reply src/cli/restore/restore.go:47]: I would use the more explicit value in the example since it is more consistent with the in-place option

**Comment by ryanfkeepers** [comment]
@gmatev ready for a second review.  I've added all the asks except for the addition to `Usage`, because I'm not quite sure where the best place to add that will be.

**Comment by vkamra** [comment]
> @gmatev ready for a second review. I've added all the asks except for the addition to `Usage`, because I'm not quite sure where the best place to add that will be.

@ryanfkeepers - can add it under a "Restore Options" page under Usage

image

**Review by vkamra — APPROVED** [review]
LGTM. Approving so we can merge and iterate.

File: website/docs/setup/advanced-restore.md
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:36]:
```suggestion
### Limitations
```
Known issues makes it confusing with the overall Known issues

File: website/docs/setup/advanced-restore.md
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:52]:
```suggestion
When restoring data into an existing folder, the items restored may conflict
```

File: website/docs/setup/advanced-restore.md
Comment (gmatev) [reply website/docs/setup/advanced-restore.md:32]: You have a version of what I'm suggesting around collisions below

File: website/docs/setup/advanced-restore.md
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:79]: Should include an example of the name modifiers when this happens.

File: website/docs/setup/advanced-restore.md
Comment (gmatev) [inline website/docs/setup/advanced-restore.md:73]:
```suggestion
The default behavior. When a collision is identified, the item is skipped and no restore is attempted.
```

**Review by gmatev — CHANGES_REQUESTED** [review]
@ryanfkeepers added some suggestions

File: website/docs/setup/advanced-restore.md
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:38]: We can make it error if the destination looks hierarchical.  That's not difficult.

Comment (gmatev) [reply website/docs/setup/advanced-restore.md:38]: That seems like a good idea to me

File: website/docs/setup/advanced-restore.md
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:48]: Done.  @gmatev Thanks for the review, second draft is ready to read.

Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:48]: Done.  @gmatev Thanks for the review, second draft is ready to read.

**Review by gmatev — APPROVED** [review]

File: website/docs/setup/advanced-restore.md
Comment (ryanfkeepers) [reply website/docs/setup/advanced-restore.md:54]: Will do in a follow-up PR.

**Pull request merged**
